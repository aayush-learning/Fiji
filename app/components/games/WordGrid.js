import React, { Component } from 'react';
import PropTypes from 'prop-types'
import TileGrid from './TileGrid'

const SIZE = 3

export default class WordGrid extends Component {
    constructor(props) {
        super(props)
        this.state = this._initBoard(props)
      }
       
  _initBoard = (props) => {
      const Data1 = props.data.word 
      const shuffledData1 = props.data.others
      .map((a, i) => [Math.floor(i / (SIZE * SIZE)), a])
      .sort((a, b) => a[0] - b[0])
      .map((a) => a[1])
 
      const letters = new Array(SIZE*SIZE)
    for (let i = 0; i < letters.length; i++) {
      letters[i] = 'null';
    }
  let h=1;
  let rno = Math.floor(Math.random() * ((SIZE*SIZE)-Data1.length+1)) + 0 ;
  const rindex = new Array(Data1.length)
   // console.log("random no",rno)
    letters[rno]=Data1[0];
    rindex[0]=rno;
  for (let i = 1; i < Data1.length; i++) {
    let flag=0,flag2=0;
   for (let j =1; j <= SIZE;j++){
     if(rno==(j*SIZE-1))
     {flag=1;break;}
    }
   if(flag==1){
     rno+=SIZE;
     let p=0;
   while(p<SIZE){
      letters[rno]=Data1[i];
      rindex[h]=rno;
      h++;
      i++;p++;
      rno--;
      if(i==Data1.length){flag2=1;console.log('breaked');break;}
    }
      if(p==SIZE&&flag2==0){flag=0;rno+=SIZE;}
   }
 if(flag==0){
   rno++;
   letters[rno]=Data1[i];
   rindex[h]=rno;
   h++;
 }}
 for (let i = 0,j=0; i < SIZE*SIZE ; i++){
   if(letters[i]=='null'){
     letters[i]=shuffledData1[j];
     j++;
   }
 } 
       let statuses = new Array(SIZE * SIZE)
        for (let i = 0; i < statuses.length; i++) {
          statuses[i] = 'neutral';
        }
      let index = 0;
       return ({
          letters,
          shuffledData1,
          index,
          statuses,
          rindex
        })
      }
      componentWillReceiveProps(nextProps) {
      this.props.runIndex != nextProps.runIndex && this.setState(this._initBoard(nextProps))
      }

  render() {
  return (
      <TileGrid
        numRows={SIZE}
        numCols={SIZE}
        data={this.state.letters}
        tileColor='#24B2EA'
        edgeColor='deepskyblue'
        statuses={this.state.statuses}
        onStatusChange={this._onStatusChange}
        pressedTileColor='goldenrod'
        pressedEdgeColor='darkgoldenrod'
        textColor='#FFFFFF'
        style={{
          width: this.props.style.width,
          height: this.props.style.height
        }}
        statusStyles = {{
            neutral: {
              View: {
                backgroundColor: 'white'
              },
              Text: {
                color: 'black'
              }
            },
            selected: {
                View: {
                  backgroundColor: 'yellow'
                },
                Text: {
                  color: 'black'
                }
              }
          }}
        onPress={this._clickTile}
      />
    )
  }
  _onStatusChange(id, view, prevStatus, currentStatus) {
    console.log('onstatuschange:',id, prevStatus, currentStatus)
   // currentStatus == 'neutral' && view.zoomIn(250)
  }
   _clickTile = (id, view) => {
    const wlen = this.props.data.word.length;
    const index = this.state.index
    const rindex = this.state.rindex
    const riddata = rindex[index];
 
      if(riddata===id)
      {
        view.pulse(10).then((endState) => {
          this.setState({...this.state,
          statuses: this.state.statuses.map((val, riddata )=> {
          return id == riddata ? 'selected' : val})})
          }) 
    this.props.onScore && this.props.onScore(2) 
    this.props.setProgress && this.props.setProgress((index + 1) / wlen)
    this.setState({...this.state, index: index + 1})
            if(id===(rindex[wlen-1])){
              setTimeout( () => {
              this.setState({...this.state, index: 0, statuses: this.state.statuses.map(()=> 'neutral')})
              this.props.onEnd()   
              }, 1200);
            }
        }
         else{
         let flag=0;
         for(var i=0;i<=index;i++){
             if(id===rindex[i]){flag=1}
            }
             if(flag==0){ view.bounce(800); }
        }
    }
  }

WordGrid.propTypes = {
 data: PropTypes.object,
  runIndex: PropTypes.number,
  onScore: PropTypes.func,
  onEnd: PropTypes.func,
  setProgress: PropTypes.func
}
