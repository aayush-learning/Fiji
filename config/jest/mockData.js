import { fetchUsersFailure } from '../../app/redux/users';

const contentJson = require('../seed/content.json')
const usersJson = require('../seed/users.json')
const userAliceJson = require('../seed/user_alice.json')
const storyJson = require('../seed/story.json')

export const titles = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('storytitle:')
})

export const stories = storyJson['docs'].filter(function (doc) {
  return doc._id.startsWith('story:')
})

export const games = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('game:')
})

export const gameThemeData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('gametheme:')
})

export const stickersData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker:caterpillar:')
})

export const stickerPacksData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('sticker-pack:')
})

export const highScoresData = contentJson['docs'].filter(function (doc) {
  return doc._id.startsWith('high-score:')
})

export const story_alice_in_wonderland = storyJson['docs'].filter(function (doc) {
  return doc._id == 'story:alice-in-wonderland'
})[0]

export const users = usersJson['docs']

export const messages = userAliceJson['docs'].filter(function(doc) {
  return doc._id.startsWith('chat:mad-hatter:')
})