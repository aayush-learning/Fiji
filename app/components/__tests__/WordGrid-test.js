import 'react-native';
import React from 'react';
import WordGrid from '../games/WordGrid';
import { titles } from '../../../config/jest/mockData'
import { Dimensions } from 'react-native'
const window = Dimensions.get("window")

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <WordGrid 
    data={{
        word: ['D', 'O', 'G'],
        others: ['O']
      }}
      style = {{
        width: window.width,
        height: window.height
      }}
      onPress={{clickTile()}}
       />
  ).getInstance()
  tree._clickTile()
  expect(tree).toMatchSnapshot()
})
