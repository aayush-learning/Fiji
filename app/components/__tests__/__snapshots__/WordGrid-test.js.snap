// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "height": 339,
        "left": 18,
        "position": "absolute",
        "top": 310,
        "width": 339,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 8,
          "elevation": 8,
          "height": 339,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "grey",
          "shadowOpacity": 1,
          "shadowRadius": 8,
          "width": 339,
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 299,
          }
        }
      >
        D
      </Text>
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "height": 339,
        "left": 393,
        "position": "absolute",
        "top": 310,
        "width": 339,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 8,
          "elevation": 8,
          "height": 339,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "grey",
          "shadowOpacity": 1,
          "shadowRadius": 8,
          "width": 339,
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 299,
          }
        }
      >
        O
      </Text>
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "height": 339,
        "left": 18,
        "position": "absolute",
        "top": 685,
        "width": 339,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 8,
          "elevation": 8,
          "height": 339,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "grey",
          "shadowOpacity": 1,
          "shadowRadius": 8,
          "width": 339,
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 299,
          }
        }
      >
        O
      </Text>
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "center",
        "height": 339,
        "left": 393,
        "position": "absolute",
        "top": 685,
        "width": 339,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 8,
          "elevation": 8,
          "height": 339,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "grey",
          "shadowOpacity": 1,
          "shadowRadius": 8,
          "width": 339,
        }
      }
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 299,
          }
        }
      >
        G
      </Text>
    </View>
  </View>
</View>
`;
